packages/app-staking/src/Account/index.tsx

```diff
@@ -274,50 +311,42 @@ class Account extends React.PureComponent<Props, State> {
+    const { accountId, staking_nominators, staking_validators, freeBalance, t } = this.props;
+    const { isActiveStash, isActiveController, sessionId, stakingLedger } = this.state;
     const buttons = [];
 
-    if (!stashId) {
-      if (!controllerId) {
+    if (isActiveStash) {
+      // only show a "Bond Additional" button if this stash account actually doesn't bond everything already
+      // staking_ledger.total gives the total amount that can be slashed (any active amount + what is being unlocked)
+      if (freeBalance && stakingLedger && stakingLedger.total && (freeBalance.gt(stakingLedger.total))) {
         buttons.push(
           <Button
             isPrimary
             key='bond'
-            onClick={this.toggleBond}
-            label={t('Bond Funds')}
+            onClick={this.toggleBondExtra}
+            label={t('Bond Additional')}
           />
         );
-      } else {
-        // only show a "Bond Additional" button if this stash account actually doesn't bond everything already
-        // staking_ledger.total gives the total amount that can be slashed (any active amount + what is being unlocked)
-        if (freeBalance && staking_ledger && staking_ledger.total && (freeBalance.gt(staking_ledger.total))) {
-          buttons.push(
-            <Button
-              isPrimary
-              key='bond'
-              onClick={this.toggleBondExtra}
-              label={t('Bond Additional')}
-            />
-          );
-        }
-        // don't show the `unbond` button if there's nothing to unbond
-        // staking_ledger.active gives the amount that can be unbonded (total - what's being unlocked).
-        if (staking_ledger && staking_ledger.active && staking_ledger.active.gtn(0)) {
-          buttons.length && buttons.push(<Button.Or key='bondAdditional.or' />);
-          buttons.push(
-            <Button
-              isNegative
-              key='unbond'
-              onClick={this.toggleUnbond}
-              label={t('Unbond')}
-            />
-          );
-        }
       }
-    } else {
+
+      // don't show the `unbond` button if there's nothing to unbond
+      // staking_ledger.active gives the amount that can be unbonded (total - what's being unlocked).
+      if (stakingLedger && stakingLedger.active && stakingLedger.active.gtn(0)) {
+        buttons.length && buttons.push(<Button.Or key='bondAdditional.or' />);
+        buttons.push(
+          <Button
+            isNegative
+            key='unbond'
+            onClick={this.toggleUnbond}
+            label={t('Unbond')}
+          />
+        );
+      }
+    } else if (isActiveController) {
       const isNominating = !!staking_nominators && !!staking_nominators[0] && staking_nominators[0].length;
       const isValidating = !!staking_validators && !!staking_validators[0] && !(staking_validators[0].isEmpty);
 
+      // if we are validating/nominating show stop
       if (isValidating || isNominating) {
```

Comment:
This is always false (`isValidating` is not derived from props) so that the "stop validation" on Alice doesn't show

Reply:
Ahhh.  Wrong block. Going to split these anyway. Dangit. A complete mess there as it stands, no wonder.